File: day-1/day-1.py
import unittest

def solve_part_1(input):
    count = 0
    for idx, x in enumerate(input[1:]):
        if x > input[idx]: count += 1
    return count

def solve_part_2(input):
    count = 0
    for idx, x in enumerate(input[3:]):
        if x > input[idx]: count += 1
    return count

with open('./input.txt') as f:
    input = map(int, [line for line in f.read().splitlines()])

class Test(unittest.TestCase):
    def test_solve_part_1_input(self):
        self.assertEqual(solve_part_1(input), 1832)

    def test_solve_part_2_input(self):
        self.assertEqual(solve_part_2(input), 1858)

unittest.main()

File: day-1/day-1.rs
fn solve_part_1(input: &Vec<i64>) -> i64 {
    let mut count = 0;
    for idx in 1..input.len() {
        if input[idx] > input[idx - 1] {
            count += 1;
        }
    }
    return count;
}

fn solve_part_2(input: &Vec<i64>) -> i64 {
    let mut count = 0;
    for idx in 3..input.len() {
        if input[idx] > input[idx - 3] {
            count += 1;
        }
    }
    return count;
}

static INPUT: &str = include_str!("input.txt");
fn main() {
    let input: Vec<i64> = INPUT.lines().map(|x| x.parse().unwrap()).collect();
    println!("{}", solve_part_1(&input));
    println!("{}", solve_part_2(&input));
}
